src/core/mutatus.scala

```diff
@@ -294,21 +294,27 @@ object Decoder extends Decoder_1 {
     *  This is a suboptimal implementation, and a better solution may be possible with more work
    **/
   def dispatch[T](st: SealedTrait[Decoder, T]): Decoder[T] =
-    value => {
-      value match {
-        case entityValue: EntityValue if entityValue.get().contains("type") =>
-          val entity = entityValue.get()
-          val typeName = entity.getString("type")
-          st.subtypes.collectFirst {
-            case subtype if subtype.typeName.full == typeName =>
-              subtype.typeclass.decodeValue(entity.getValue("value"))
+    encodedValue => {
+      // Naive approach used when fetching entity not indexed by mutatus
+      def firstSucessfull(value: Value[_]) = {
+        st.subtypes.toStream
+          .map { subtype =>
+            Try(subtype.typeclass.decodeValue(value))
           }
-        case value => // Naive approach used when fetching entity not indexed by mutatus
-          st.subtypes.toStream
-            .map { subtype =>
-              Try(subtype.typeclass.decodeValue(value))
+          .collectFirst { case Success(value) => value }
+      }
+
+      encodedValue match {
+        case entityValue: EntityValue =>
+          Try(entityValue.get.getEntity("_meta").getString("typename"))
```

Comment:
One very minor suggestion: use a `final val` for `"_meta"`, just to avoid the risk of a typo.